ionoscloud/resource_cdn_distribution.go

```diff
@@ -0,0 +1,255 @@
+										Required:         true,
+										ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{"R1", "R5", "R10", "R25", "R50", "R100", "R250", "R500"}, true)),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func resourceCdnDistributionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(services.SdkBundle).CdnClient
+
+	distributionDomain := d.Get("domain").(string)
+
+	distribution := ionoscloud_cdn.DistributionCreate{
+		Properties: &ionoscloud_cdn.DistributionProperties{
+			Domain: &distributionDomain,
+		},
+	}
+
+	if attr, ok := d.GetOk("certificate_id"); ok {
+		attrStr := attr.(string)
+		distribution.Properties.CertificateId = &attrStr
+	}
+
+	if routingRules, err := cdnService.GetRoutingRulesData(d); err == nil {
+		distribution.Properties.RoutingRules = routingRules
+	} else {
+		return diag.FromErr(err)
+	}
+
+	createdDistribution, _, err := client.SdkClient.DistributionsApi.DistributionsPost(ctx).DistributionCreate(distribution).Execute()
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("error creating CDN distribution (%s) (%w)", d.Id(), err))
+		return diags
+	}
+	d.SetId(*createdDistribution.Id)
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsDistributionReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occurred while checking the status for the CDN Distribution with ID: %v, error: %w", d.Id(), err))
+	}
+
+	log.Printf("[INFO] CDN Distribution Id: %s", d.Id())
+
+	return resourceCdnDistributionRead(ctx, d, meta)
+}
+
+func resourceCdnDistributionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CdnClient
+
+	distribution, apiResponse, err := client.SdkClient.DistributionsApi.DistributionsFindById(ctx, d.Id()).Execute()
+
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		diags := diag.FromErr(fmt.Errorf("error while fetching CDN distribution %s: %w", d.Id(), err))
+		return diags
+	}
+
+	log.Printf("[INFO] Successfully retrieved CDN distribution %s: %+v", d.Id(), distribution)
+
+	if err := cdnService.SetDistributionData(d, distribution); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func resourceCdnDistributionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).CdnClient
+
+	distributionDomain := d.Get("domain").(string)
+
+	request := ionoscloud_cdn.DistributionUpdate{
+		Id: ionoscloud_cdn.ToPtr(d.Id()),
+		Properties: &ionoscloud_cdn.DistributionProperties{
+			Domain: &distributionDomain,
+		},
+	}
+
+	if attr, ok := d.GetOk("certificate_id"); ok {
+		attrStr := attr.(string)
+		request.Properties.CertificateId = &attrStr
+	}
+
+	if routingRules, err := cdnService.GetRoutingRulesData(d); err == nil {
```

Comment:
same here

Reply:
Done

Reply:
Changing this required creating a new variable outside the the if so I decided to undo it